controller/embedding/manager.py

```diff
@@ -1,18 +1,18 @@
 import time
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Optional
 
 from submodules.model import enums
 from util import daemon
 from . import util
 from . import connector
-from controller.misc import manager as misc
 from controller.model_provider import manager as model_manager
-from submodules.model.business_objects import attribute
+from controller.project import manager as project_manager
+from submodules.model.business_objects import attribute, embedding, agreement, general
 
 
-def get_recommended_encoders() -> List[Any]:
+def get_recommended_encoders(is_managed: bool) -> List[Any]:
     recommendations = connector.request_listing_recommended_encoders()
-    if misc.check_is_managed():
+    if is_managed:
```

Comment:
Since the model provider doesn't exist for os we would run into an error if someone would call the function with true even though it's actually not managed. so imo the parameter would be a "false/faulty" option for a dev. Since the user of the function can't actually decide the call is inside the function.

now the misc call is done in the line before the function.

the functionality for the existing case wasn't changed so why was it changed to a parameter? Both approaches should work but I'm still curious :D

Reply:
wrote comment